src/targets/__tests__/symbolCollector.test.ts

```diff
@@ -0,0 +1,138 @@
+import { withTempDir } from '../../utils/files';
+import { NoneArtifactProvider } from '../../artifact_providers/none';
+import { checkExecutableIsPresent, spawnProcess } from '../../utils/system';
+import { SymbolCollector, SYM_COLLECTOR_BIN_NAME } from '../symbolCollector';
+
+jest.mock('../../utils/files');
+jest.mock('../../utils/system');
+jest.mock('fs', () => {
+  const original = jest.requireActual('fs');
+  return {
+    ...original,
+    promises: {
+      mkdir: jest.fn(() => {
+        /** do nothing */
+      }),
+    },
+  };
+});
+
+const customConfig = {
+  batchType: 'batchType',
+  bundleIdPrefix: 'bundleIdPrefix-',
+};
+
+function getSymbolCollectorInstance(
+  config: Record<string, unknown> = { testKey: 'testVal' }
+): SymbolCollector {
+  return new SymbolCollector(
+    {
+      name: 'symbol-collector',
+      ...config,
+    },
+    new NoneArtifactProvider()
+  );
+}
+
+describe('target config', () => {
+  test('symbol collector not present in path', () => {
+    (checkExecutableIsPresent as jest.MockedFunction<
+      typeof checkExecutableIsPresent
+    >).mockImplementationOnce(() => {
+      throw new Error('Checked for executable');
```

Comment:
Would have been better to make the executable path configurable and make sure we actually throw on a missing file.